### DISCUSSION TITLE
Announcement of the start of the update v3.2.0

### DISCUSSION THREAD

**Discussion by mrghozzi** [mrghozzi/myads#66] [Announcements]
[v3.2.0](https://github.com/mrghozzi/myads/tree/v3.2.0)
**Fix**: Multiple security bugs.
**Delete**: Share and login from wasp.gq
**Add**: Share via telegram
**Add**: Added `<nav>` to divide posts in the member's profile
**Add**: Friends page (following from both sides)
**Improvement**: Multi-language system
**Fix**: Update system to the latest version
**Fix**: Multiple page issue in the forum
**Improvement**: Visit Exchange System
**Add**: Auto-copy feature when clicking on text boxes in `b_code.php`, `l_code.php` and `r_code.php` pages
**Improvement**: message list
**Add**: notifications when ads are blocked, enabled, or removed
**Add**: a copy link button for community posts
**Add**: The ability to publish works in the store in exchange for points (PTS).
**Improvement**: UI and security in state and template files

**Comment by alaa-eg** [comment]
يرجى حل مشكلة اللغة العربية والكتابة من اليمين إلى اليسار في مواضيع المنتدى وفي صندوق الكتابة

إذا نسخت موضوع في موقع آخر وحاولت تنسيقة بتوسيط الكتابة أو جعلها من اليمين إلى اليسار لا تعمل

**Reply by mrghozzi** [reply]
مرحبا 

بالنسبة لمحرر الكتابة جرب RTL

https://i31.servimg.com/u/f31/17/77/42/11/firefo10.png

و بالنسبة لملف robots.txt
يبدو أنك قمت بتعديل في ملف .htaccess
سأبحث عن حل

**Reply by alaa-eg** [reply]
> مرحبا
> بالنسبة لمحرر الكتابة جرب RTL
> 
> https://i31.servimg.com/u/f31/17/77/42/11/firefo10.png
> 
تم حل مشكلة robots.txt

تم تجربته بالنسبة لمحرر الكتابة جرب RTL ولم يعمل جيداً

**Comment by alaa-eg** [comment]
**في الموبايل :** اسم المنتديات والمواضيع تظهر كل كلمة في سطر
https://adstn.is-best.net/forum

لو كان اسم المنتديات ولمواضيع يظهر بالكامل في نفس السطر وإذا كان طويل ينزل للسطر الثاني

يمكن عمل هذا بجعل اسم المنتدى أولأ وفي السطر الذي بعده يظهر اسم آخر مشاركة به

مثال
https://i41.servimg.com/u/f41/15/54/04/99/110.png

**Reply by mrghozzi** [reply]
سأعمل على حل مؤقت في التحديث القادم إلى حين إعادة بناء قالب المنتدى لاحقا ان شاء الله

التحديث القادم سيكون مركز على جملة من الإصلاحات في البرمجة و هيكلة الملف الشخصي للعضو 

هذا مثال على العمل الحالي https://i31.servimg.com/u/f31/17/77/42/11/firefo11.png

**Comment by alaa-eg** [comment]
**اقتراح : اضافة امكانية تعديل ترتيب "أقسام دليل المواقع"**
- إمكانيه حذف لغة من الموقع والإبقاء على لغة واحده فقط

**Reply by mrghozzi** [reply]
تعديل ترتيب أقسام دليل المواقع تم إضافتها منذ التحديث السابق v3.1.0 

![img](https://i31.servimg.com/u/f31/17/77/42/11/firefo12.png)

بالنسبة لخيار اللغة الوحيدة ليس من الصعب إضافة هذا الخيار لكن الإشكال حاليا  جميع المصطلحات في الموقع غير متاحة لخيار تعدد اللغات .
لكن سأضع هذا المقترح في جدول أعمالي إن لم يكن متاح في التحديث القادم سأعمل عله في أول تحديث لاحق

**Comment by alaa-eg** [comment]
**- اقتراح اضافة خاصية defer أو async لكود الاعلانات البنرات والنصي**

و loading="lazy" لكل الصور

**Reply by mrghozzi** [reply]
سيطلب الأمر إعادة بناء ملف bn.php 
على العموم من الممكن أن يكون التحديث القادم هو الأخير 
(البرنامج مفتوح المصدر بإمكانك تطويره في المستقبل)

**Comment by mrghozzi** [comment]
- Added: Added <nav> to divide posts in the member's profile
- Added: Friends page (following from both sides)
- Some fixes

https://github.com/mrghozzi/myads/commit/791f1050f56af52b4a9da91ba6943e596401c081

**Comment by alaa-eg** [comment]
هناك مشكلة في أقسام المنتدى عند الضغط على رقم أي صفحة لمشاهدة باقي المواضيع يتم إعادة فتح الصفحة الأولى فقط

**Reply by mrghozzi** [reply]
شكرا على الملاحظة
تم الإصلاح
إستبدل محتو ملف` /forum.php`
بهذا
https://github.com/mrghozzi/myads/blob/v3.2.0/forum.php
